src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -210,6 +210,56 @@ String decrypt(final String encryptedCredential) {
         return new String(decryptedResult.getResult(), StandardCharsets.UTF_8);
     }
 
+    /**
+     * Removes the credential fields from a template
+     * @param template the template
+     * @return the redacted template
+     */
+    public Template redactTemplateCredentials(Template template) {
+        Template.Builder processedTemplateBuilder = new Template.Builder();
+
+        Map<String, Workflow> processedWorkflows = new HashMap<>();
+        for (Map.Entry<String, Workflow> entry : template.workflows().entrySet()) {
+
+            List<WorkflowNode> processedNodes = new ArrayList<>();
+            for (WorkflowNode node : entry.getValue().nodes()) {
+                if (node.userInputs().containsKey(CREDENTIAL_FIELD)) {
```

Comment:
This seems like a special case by field name.  We should probably do this generally for _all_ encrypted fields.  This is fine for now (since we only have one encrypted field) but let's add an issue to generalize the "special keys" in this method and `processTemplateCredentials()` to condition on a list of special-case fields instead of just this one.  (There's currently a TODO stating exactly that.)